filters/private/NormalResult.hpp

```diff
@@ -0,0 +1,80 @@
+#include <Eigen/Dense>
+
+#include <pdal/KDIndex.hpp>
+
+namespace pdal
+{
+
+using namespace Eigen;
+
+struct NormalResult
+{
+    struct error : std::runtime_error
+    {
+        error(const std::string& err) : std::runtime_error(err)
+        {}
+    };
+
+    NormalResult() : m_normal(Vector3d::Zero()), m_curvature(0.0)
+    {}
+
+    Vector3d m_normal;
+    double m_curvature;
+
+    bool empty() const { return m_normal.isZero(); }
+
+    void calcNormal(PointView& view, PointIdList neighbors)
```

Comment:
I'm not a fan of throwing exceptions like this. It obligates every user to catch and they may not even be aware that this function throws. I can't think of any reason not to just return some status.

The only places we throw that I'm aware of is when the use is entirely internal to something local and it would be really cumbersome to return an error.